.github/workflows/integration_test.yml

```diff
@@ -8,7 +8,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        kube_provider: [kind]
+        kube_provider: [kind, k3d]
```

Comment:
I don't think we need to test in both tools... in the end they both create a similar k8s cluster.
Just one type of tool in CI should be enough...
Testing creating cluster with both tools is just an additional overhead that might not be necessary...

Support for more cluster types is helpful for the developer.